### PR TITLE
Fix UI inconsistency in vSphere config section when in view mode

### LINKED ISSUE #10181: vSphere's Config Page (View Mode) Has Misleading Display of Enabled Buttons/Checkboxes
**Setup**
- Rancher version: 2.8-head

**Describe the bug**

When viewing your vSphere Cluster Configuration Page in the view mode you should not be able to edit the configuration. Though Save button is not shown on this page, having some buttons/checkboxes enabled and some disabled is inconsistent and confusing.

Note: The Cluster Configuration section is relevant to all the supported providers and is covered by [#10182](https://github.com/rancher/dashboard/issues/10182).

**To Reproduce**

Create a vSphere cluster and go to its Config page.

**Screenshots**

image

image

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
Fixes #10181 

### Occurred changes and/or fixed issues

Taking the view mode into account when disabling vSphere's config form.

### Screenshot/Video

image

### CODE REVIEW

**Comment by momesgin** [comment]
@rak-phillip Unfortunately the logic inside the `ArrayList` (which is used by `ArrayListSelect` here) to show the disabled UI is not consistent with the `KeyValue`'s logic, this leads to showing the Network/Tags labels without the disabled button.

File: shell/assets/translations/en-us.yaml [hunk 1]
```diff
@@ -1680,7 +1680,7 @@ cluster:
   banner:
     warning: 'This cluster contains a machineSelectorConfig, which this form does not fully support; use the YAML editor to manage the full configuration.'
     os: 'You are attempting to add a {newOS} worker node to a cluster with one or more {existingOS} worker nodes: some installed apps may need to be upgraded or removed.'
-    rke2-k3-reprovisioning: 'Making changes to cluster configuration may result in nodes reprovisioning. For more information see the <a target="blank" href="{docsBase}/cluster-provisioning/rke-clusters/behavior-differences-between-rke1-and-rke2/" target="_blank" rel="noopener nofollow">documentation</a>.'
+    rke2-k3-reprovisioning: 'Making changes to cluster configuration may result in nodes reprovisioning. For more information see the <a href="{docsBase}/cluster-provisioning/rke-clusters/behavior-differences-between-rke1-and-rke2/" target="_blank" rel="noopener nofollow">documentation</a>.'
```
Comment (rak-phillip) [inline shell/assets/translations/en-us.yaml:1683]: Does this change belong in this PR? I'm not against it in any way, it just appears to be unrelated to other than he's represented here.

**Review by rak-phillip — COMMENTED** [review]
I think this looks good, I just have one question before we move on.

File: shell/assets/translations/en-us.yaml (hunk 1 above)
Comment (momesgin) [reply shell/assets/translations/en-us.yaml:1683]: I didn't find this particular change to deserve its own ticket, but I'm totally fine with excluding it

Comment (momesgin) [reply shell/assets/translations/en-us.yaml]: done

**Comment by aalves08** [comment]
@momesgin needs a rebase because of conflicts

**Review by rak-phillip — APPROVED** [review]
